url5a01f2c1d9_core9_core_tgz-pJ8-io_core9_core_boot_BootstrapFrameworkJ8/src/io/core9/core/commands/Command.java

```diff
@@ -0,0 +1,43 @@
+package io.core9.core.commands;
+
+public abstract class Command {
+	
+	private String className;
+	private String method;
+	private String type;
+	private Integer priority;
+	
+	public String getClassName() {
+		return className;
+	}
+	public void setClassName(String className) {
+		this.className = className;
+	}
+	public String getMethod() {
+		return method;
+	}
+	public void setMethod(String method) {
+		this.method = method;
+	}
+	public String getType() {
+		return type;
+	}
+	public void setType(String type) {
+		this.type = type;
+	}
+	public Integer getPriority() {
+		return priority;
+	}
+	public void setPriority(Integer priority) {
+		this.priority = priority;
+	}
+	
+	public abstract Object[] execute(Object... args);
+	
+	public Command(String name, Integer priority) {
+		this.className = name.substring(0, name.indexOf(':'));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

If the indexOf() search on line 38 does not find the substring and returns -1, the call to substring() on line 38 will throw IndexOutOfBoundsException. Note that the method is publicly accessible to external callers who might not be aware of the constraint on string name.